scripts/west_commands/sign.py

```diff
@@ -173,11 +173,6 @@ def do_run(self, args, ignored):
         elif args.gen_hex is None and hex_exists:
             formats.append('hex')
 
-        if not formats:
```

Comment:
I understand the reason for not having this for RimageSigner, but then it should probably be moved down to the
ImgtoolSigner code, and not simply removed.

@mbolivar-nordic comments ?

Reply:
@tejlmand: rimage aside, do you understand how silently building nothing is a valid use case?

Reply:
Yeah, that was my thinking with just removing this.  Basically "If we really cared about detection and reporting of this particular error condition, we would have been louder about it".  To my eyes a python stack dump to the terminal would have actually been preferable.

Reply:
@marc-hb  I was more thinking that @mbolivar-nordic probably had a reason to do this in first place (and probably for ImgtoolSigner), and as @andyross mention:
> Just present the possibly-empty list of output formats to the Signer
object and let it make the decision about whether an empty formats
list is an error.

Code was changed specifically for Rimage, and I wanted to be sure that ImgtoolSigner object wasn't forgotten by mistake.
That is, should have the message there instead.

Reply:
I'm afraid you just confused me even more.

The code that was removed does not really seem specific to either `ImgtoolSigner` or `RimageSigner` because it was effectively making sure none of them ever ran at all in some specific conditions.

The code just removed was added as part of the larger commit d371c54f5182 which is a sanity check on missing input files. It's actually a bit hard to picture the entire "Truth Tables" because the boolean flags have actually 3 states (with, without, default) and then this is all combined with checks for input files and Kconfig options. If nothing else removing this code makes these truth tables slightly smaller - no one should regret that :-)